extensions-core/kinesis-indexing-service/src/main/java/org/apache/druid/indexing/kinesis/supervisor/KinesisSupervisor.java

```diff
@@ -232,39 +231,34 @@ protected void scheduleReporting(ScheduledExecutorService reportingExec)
   @Override
   protected int getTaskGroupIdForPartition(String partitionId)
   {
-    if (!partitionIds.contains(partitionId)) {
-      partitionIds.add(partitionId);
-    }
-
     return getTaskGroupIdForPartitionWithProvidedList(partitionId, partitionIds);
   }
 
   private int getTaskGroupIdForPartitionWithProvidedList(String partitionId, List<String> availablePartitions)
   {
+    int index = availablePartitions.indexOf(partitionId);
+    if (index < 0) {
+      return index;
+    }
     return availablePartitions.indexOf(partitionId) % spec.getIoConfig().getTaskCount();
   }
 
   @Override
-  protected Map<Integer, ConcurrentHashMap<String, String>> recomputePartitionGroupsForExpiration(
+  protected Map<Integer, Set<String>> recomputePartitionGroupsForExpiration(
```

Comment:
super nit: i think this fits on one line now. Unrelated i also see an unused static hash function from previous attempts at fixing all the holes in this sharding stuff..

Reply:
Changed to one line, removed the unused hash fn